lib/board.rb

```diff
@@ -0,0 +1,55 @@
+require 'pry'
+
+class Board
+  attr_accessor :cells
+
+  def initialize
+	  reset! 
+  end
+
+  #Reset the arrary to 9 blank strings
+	def reset!
+		@cells = Array.new(9, " ")
+	end
+
+  #Display the board with blank strings
+  def display
+  	puts "       #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "       -----------"
+    puts "       #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "       -----------"
+    puts "       #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  #Find an element at certain index in the @cells array
+  def position(index)
+    @cells[index.to_i-1] 
+  end
+
+  #Check if the entier @cells array occupied 
+  def full?
+  	@cells.all?{ |cell| cell == "X" || cell == "O"}
+  end
+
+  #Count the number of turns that has been played
+  def turn_count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```